index.bs

```diff
@@ -10093,55 +10090,65 @@ Dictionary {{AudioWorkletNodeOptions}} Members</h6>
-		to initialize {{AudioParam}} objects in
-		{{AudioWorkletNode}}. If the string key of an entry in the
-		list does not match the name of any {{AudioParam}} objects in
-		the node, it is ignored.
+		to set the initial {{AudioParam/value}} of an {{AudioParam}}
+		with the matched name in the {{AudioWorkletNode}}.
 
 	: <dfn>processorOptions</dfn>
 	::
-		This holds any additional user-defined data that may be used to
-		initialize the corresponding {{AudioWorkletProcessor}} for this
-		{{AudioWorkletNode}}.
+		This holds any user-defined data that may be used to initialize
+		custom properties in an {{AudioWorkletProcessor}} instance
+		that is associated with the {{AudioWorkletNode}}.
 </dl>
 
 <h6 id="configuring-channels-with-audioworkletnodeoptions">
 Configuring Channels with {{AudioWorkletNodeOptions}}</h6>
 
-With a combination of {{AudioWorkletNodeOptions/numberOfInputs}},
-{{AudioWorkletNodeOptions/numberOfOutputs}},
-and {{AudioWorkletNodeOptions/outputChannelCount}},
-various channel configurations can be achieved.
-
-: {{AudioWorkletNodeOptions/numberOfInputs}} = 0,
-	{{AudioWorkletNodeOptions/numberOfOutputs}} = 0
-:: {{NotSupportedError}} MUST be thrown by the constructor.
-
-: {{AudioWorkletNodeOptions/numberOfInputs}} = 1,
-	{{AudioWorkletNodeOptions/numberOfOutputs}} = 1
-:: If {{AudioWorkletNodeOptions/outputChannelCount}} is
-	unspecified, the output channel count will match
-	<a>computedNumberOfChannels</a> from the input. If
-	{{AudioWorkletNodeOptions/outputChannelCount}} is specified
-	with one value, the output channel count will match the given value.
-
-: All other cases
-:: If {{AudioWorkletNodeOptions/outputChannelCount}} is
-	unspecified, it will be mono for all outputs. If specified,
-	the values will be used as given.
+The following algorithm describes how an {{AudioWorkletNodeOptions}} can be
+used to configure various channel configurations.
+
+<div id="configure-with-audioworkletnodeoptions" algorithm="configure with AudioWorkletNodeOptions">
+	1. Let <var>node</var> be an {{AudioWorkletNode}} instance that is
+		given to this algorithm.
+
+	1. If both {{AudioWorkletNodeOptions/numberOfInputs}} and
+		{{AudioWorkletNodeOptions/numberOfOutputs}} are zero,
+		throw a {{NotSupportedError}} and abort the remaining steps.
+
+	1. If the length of {{AudioWorkletNodeOptions/outputChannelCount}}
+		does not match {{AudioWorkletNodeOptions/numberOfOutputs}},
+		throw an {{IndexSizeError}} and abort the remaining steps.
+
+	1. If any value in {{AudioWorkletNodeOptions/outputChannelCount}}
+		is zero or greater than the implementation’s maximum number of
+		channels, throw a {{NotSupportedError}} and abort the remaining
+		steps.
+
+	1. If both {{AudioWorkletNodeOptions/numberOfInputs}} and
+		{{AudioWorkletNodeOptions/numberOfOutputs}} are 1,
+
+		1. If {{AudioWorkletNodeOptions/outputChannelCount}} is
+			given and it contains a value, the channel count of the
+			<var>node</var> output will be set to the value and
```

Comment:
The "will be set to ABC and return" phrasing on all of these is a little weird.  It would be clearer to say things like "set the channel count of the node output to ABC and return".